src/exception.jl

```diff
@@ -44,6 +49,37 @@ function show(io::IO, e::PyError)
     end
 end
 
+function pyerror(msg::AbstractString)
+        ptype, pvalue, ptraceback = Ref{PyPtr}(), Ref{PyPtr}(), Ref{PyPtr}()
+        # equivalent of passing C pointers &exc[1], &exc[2], &exc[3]:
+        ccall((@pysym :PyErr_Fetch), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        ccall((@pysym :PyErr_NormalizeException), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        pyerror(msg, PyObject(ptype[]), PyObject(pvalue[]), PyObject(ptraceback[]))
+end
+
+function pyerror(msg::AbstractString, e::PyError)
+    pyerror(msg, e.T, e.val, e.traceback)
+end
+
+function pyerror(msg::AbstractString, ptype::PyObject, pvalue::PyObject, ptraceback::PyObject)
+    pargs = _getproperty(pvalue, "args")
+
+    # If the value of the error is a PyJlError, it was generated in a pyjlwrap callback, and
+    # we forward it.
+    if pargs != C_NULL
+        args = PyObject(pargs)
+        if length(args) > 0
+            arg = PyObject(@pycheckn ccall((@pysym :PySequence_GetItem), PyPtr, (PyPtr,Int), args, 0))
```

Comment:
You'll need to move up the definition of the `@pycheck*` macros in this file.

Reply:
I missed your comment!.
I just moved down pyerror() but I can swap them if you prefer.